src/devices/sound/k054539.cpp

```diff
@@ -285,9 +322,12 @@ void k054539_device::sound_stream_update(sound_stream &stream, std::vector<read_
 					LOG("Unknown sample type %x for channel %d\n", base2[0] & 0xc, ch);
 					break;
 				}
-				lval += cur_val * lvol;
-				rval += cur_val * rvol;
-				rbase[(rdelta + reverb_pos) & 0x1fff] += int16_t(cur_val*rbvol);
+
+				const float filter_val = calculate_filter(ch, filter_frac);
+
+				lval += filter_val * lvol;
+				rval += filter_val * rvol;
+				rbase[(rdelta + reverb_pos) & 0x1fff] += int16_t(filter_val*rbvol);
```

Comment:
Sorry for being so late, but I just noticed that this PR makes the reverb not work anymore.

Previously, the range of `cur_val` was ±32768, which worked nicely with the `int16_t rbase[]` buffer.
Now, the range of `filter_val` is ±1.0, so the reverb is always going to be truncated to 0 due to the typecast from float to int.

A possible workaround I could imagine would be to apply a scale of 32768.0 when reading/writing `rbase`.